state/web3.go

```diff
@@ -503,6 +508,19 @@ func (w *Web3) calcPartialBalances(hc HoldersCandidates, currentLog types.Log) (
 		} else {
 			hc[logData.From] = big.NewInt(-1)
 		}
+	case CONTRACT_TYPE_ERC721_BURNED:
+		filter := w.contract.(*erc721.ERC721Contract).ERC721ContractFilterer
+		logData, err := filter.ParseTransfer(currentLog)
+		if err != nil {
+			return hc, err
+		}
+		if logData.To == common.HexToAddress(NULL_ADDRESS) {
+			if fromBalance, exists := hc[logData.From]; exists {
+				hc[logData.From] = new(big.Int).Add(fromBalance, big.NewInt(1))
```

Comment:
What happens if the transaction is burning more than 1 NFTs? Is this a possibility? Shouldn't we just take the amount that is being burned and add it to the holder address?

Reply:
In the ERC721 standard the burn process cannot be batched. So an event will be emitted per burn.